src/main/java/net/dv8tion/jda/internal/entities/TextChannelImpl.java

```diff
@@ -407,7 +407,7 @@ public MessageAction sendFile(@Nonnull byte[] data, @Nonnull String fileName, @N
     @Override
     public RestAction<Void> pinMessageById(@Nonnull String messageId)
     {
-        checkPermission(Permission.MESSAGE_READ, "You cannot pin a message in a channel you can't access. (MESSAGE_READ)");
+        checkPermission(Permission.VIEW_CHANNEL, "You cannot pin a message in a channel you can't access. (MESSAGE_READ)");
```

Comment:
These error messages still reference `MESSAGE_READ`.
Please replace all references to `MESSAGE_READ` throughout the code.